include/roaring/roaring_array.h

```diff
@@ -95,6 +95,11 @@ inline container_t *ra_get_container_at_index(
  */
 uint16_t ra_get_key_at_index(const roaring_array_t *ra, uint16_t i);
 
+/**
+ * Check whether value should be placed in the container at index i
+ */
+inline bool ra_test_value_at_index(const roaring_array_t *ra, uint16_t i, uint32_t value);
```

Comment:
This `inline`/`extern inline` is backwards (and should emit a warning: "inline function 'ra_test_value_at_index' is not defined". The definition should be here (otherwise, how can it inline?), and the `extern inline` in the `.c` file should be a declaration only:
```c
// In the header
inline bool ra_test_value_at_index(..) {
  // body here
}

// In the .c file, this will cause the single real instantiation of the function
extern inline bool ra_test_value_at_index(..);
```

Alternatively, just define the function as `static inline` in the header, and don't mention it in the `.c` file at all.

Reply:
that's reasonable, it should be `static inline`
`extern inline` makes me feel confused about how it would be compiled by the compiler, unfold or just marked as an outer symbol for calling.

Reply:
> This `inline`/`extern inline` is backwards (and locally for me emits a warning: "inline function 'ra_test_value_at_index' is not defined"). The definition should be here (otherwise, how can it inline?), and the `extern inline` in the `.c` file should be a declaration only:
> 
> ```c
> // In the header
> inline bool ra_test_value_at_index(..) {
>   // body here
> }
> 
> // In the .c file, this will cause the single real instantiation of the function
> extern inline bool ra_test_value_at_index(..);
> ```
> 
> Alternatively, just define the function as `static inline` in the header, and don't mention it in the `.c` file at all.

But you can find some `extern inline` in this repo... https://github.com/RoaringBitmap/CRoaring/issues/325

Reply:
@Dr-Emann I have questioned about this before.